server/api.js

```diff
@@ -1,14 +1,118 @@
+import { request } from "express";
 import { Router } from "express";
 import db from "./db";
 
 import logger from "./utils/logger";
 
 const router = Router();
 
+
+let fakeLaptopAssign = [
+	{
+		laptopId: 1,
+		laptopName: "Dell",
+		deliveryOption: "pickup",
+		address: "donator address: 123 main st",
+	}, 
+	{
+		laptopId: 2,
+		laptopName: "Mac",
+		deliveryOption: "ship",
+		address: "donator address: 124 main st",
+	},
+];
+
+let fakeRequests = [
+	{
+		id: 100,
+		firstName: "bob",
+		lastName: "the builder",
+		email: "email@email.com",
+		phoneNumber: "073820384924",
+		status: "WAITING",
+		laptopAssignment: {},
+		requestAddress: "",
+	},
+	{
+		id: 110,
+		firstName: "hulk",
+		lastName: "the smasher",
+		email: "email@email.com",
+		phoneNumber: "073820384924",
+		status: "ASSIGNED",
+		laptopAssignment: fakeLaptopAssign[1],
+		requestAddress: " users address : 123 south street",
+	},
+	{
+		id: 111,
+		firstName: "tony",
+		lastName: "stack",
+		email: "email@email.com",
+		phoneNumber: "073820384924",
+		status: "ASSIGNED",
+		laptopAssignment: fakeLaptopAssign[0],
+		requestAddress: "users address : 123 south street",
+	},
+	{
+		id: 120,
+		firstName: "silver",
+		lastName: "surfer",
+		email: "email@email.com",
+		phoneNumber: "073820384924",
+		status: "ASSIGNED",
+		laptopAssignment: fakeLaptopAssign[1],
+	},
+
+	{
+		id: 120,
+		firstName: "silver",
+		lastName: "surfer",
+		email: "email@email.com",
+		phoneNumber: "073820384924",
+		status: "ASSIGNED",
+		laptopAssignment: fakeLaptopAssign[0],
+	},
+
+	{
+		id: 130,
+		firstName: "hulk",
+		lastName: "the smasher",
+		email: "email@email.com",
+		phoneNumber: "073820384924",
+		status: "CANCELLED",
+		laptopAssignment: {},
+		requestAddress: "",
+	},
+	{
+		id: 140,
+		firstName: "Don",
+		lastName: "Yen",
+		email: "email@email.com",
+		phoneNumber: "073820384924",
+		status: "FULLFILLED",
+	},
+];
+
+router.get("/laptop-donation/:id", async (req, res) => {
+	 console.log("here");
+		try {
+			const result = await db.query("SELECT * from laptop_donation WHERE id = $1", [req.params.id]);
+			res.json(result.rows[0]);
```

Comment:
the mapping to camelCase names needs to happen on result.rows

Reply:
corrected